resources/asciidoctor/lib/inline_callout/extension.rb

```diff
@@ -0,0 +1,70 @@
+require 'asciidoctor'
+
+include Asciidoctor
+
+##
+# Enables inline callouts which asciidoc supports but asciidoctor doesn't.
+# Filed as enhancement request at
+# https://github.com/asciidoctor/asciidoctor/issues/3037
+#
+# Usage
+#
+#   POST <1> /_search/scroll <2>
+#
+# NOTE: This isn't an asciidoctor extension so much as a hack. Just including
+# the file causes us to hack asciidoctor to enable the behavior. By default we
+# don't do anything if you don't set the the `inline-callouts` attribute so
+# you need to *ask* for the change in behavior.
+#
+module InlineCallout
+  InlineCalloutScanRx = /\\?<!?(|--)(\d+|\.)\1>/
+  InlineCalloutSourceRx = %r(((?://|#|--|;;) ?)?(\\)?&lt;!?(|--)(\d+|\.)\3&gt;)
+  InlineCalloutSourceRxt = "(\\\\)?&lt;()(\\d+|\\.)&gt;"
+  InlineCalloutSourceRxMap = ::Hash.new {|h, k| h[k] = /(#{::Regexp.escape k} ?)?#{InlineCalloutSourceRxt}/ }
+
+  # Disable VERBOSE so we don't log any warnings. It really isn't great to
+  # have to patch these in like this but it gets the job done and we're looking
+  # to get this into Asciidoctor proper. These methods are basically the same
+  # as the methods in asciidoctor but with new regexes.
+  old_verbose = $VERBOSE
+  $VERBOSE = false
+  Parser.class_eval do
+    def self.catalog_callouts(text, document)
+      found = false
+      autonum = 0
+      callout_rx = (document.attr? 'inline-callouts') ? InlineCalloutScanRx : CalloutScanRx
+      text.scan(callout_rx) {
+        # lead with assignments for Ruby 1.8.7 compat
```

Comment:
I doubt 1.8.7 is going to be a concern when asciidoctor requires >= 2.3

And if the answer is that we use 1.8.7 somewhere then I'm going to go ahead now and ask what our plan is to stop that.

Reply:
Honestly I like the line even without 1.8.7 because it is easier to read. I think, at least.